src/main/java/session/MenuSession.java

```diff
@@ -127,8 +130,26 @@ private void initializeBreadcrumbs() {
         this.breadcrumbs.add(SessionUtils.getAppTitle());
     }
     
-    public void updateApp(String updateMode) {
-        this.engine.attemptAppUpdate(updateMode);
+    public NotificationMessage updateApp(String updateMode) {
+        try {
+            if (this.engine.attemptAppUpdate(updateMode)) {
+                return new NotificationMessage("Application updated successfully.", false);
+            } else {
+                return new NotificationMessage("Application up to date.", false);
+            }
+        } catch (UnresolvedResourceException e) {
+            String message = "Update Failed! Couldn't find or install one of the remote resources";
+            log.error(message, e);
+            return new NotificationMessage(message, true);
+        } catch (UnfullfilledRequirementsException e) {
+            String message = "Update Failed! This CLI host is incompatible with the app";
```

Comment:
This seems like a strange message to show in formplayer; I don't think we want to mention the CLI here

Reply:
Good call